File: backend/controllers/docController.mjs
import * as docModel from '../models/docModel.mjs';

export const getAllDocs = async (req, res) => {
  try {
    const docs = await docModel.getAllDocs();
    res.json(docs);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

export const getDocById = async (req, res) => {
  try {
    const doc = await docModel.getDocById(req.params.id);
    if (!doc) {
      return res.status(404).json({ error: 'Document not found' });
    }
    res.json(doc);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

export const getSectionsForDocument = async (req, res) => {
  try {
    const sections = await docModel.getSectionsForDocument(req.params.id);
    res.json(sections);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

export const getAllSections = async (req, res) => {
  try {
    const sections = await docModel.getAllSections();
    res.json(sections);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

export const getLinksForDocument = async (req, res) => {
  try {
    const links = await docModel.getLinksForDocument(req.params.id);
    res.json(links);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};


export const newLink = async (req, res) => {
  try {
    const source_doc_id = req.body.sourceDoumentId
    const source_section_id = req.body.sourceSectionId
    const target_doc_id = req.body.targetDoumentId
    const target_section_id = req.body.targetSectionId 
    const link_type = req.body.linkType
    const created_by = req.body.createdBy 


    const response = await docModel.setLink(source_doc_id,source_section_id,target_doc_id,target_section_id,link_type,created_by);
    console.log(response)
    console.log("hello)")
    //get necessary ingo
    //create Link = new Link()
    //save link in db
    //return link
    res.json(response);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
};

File: backend-flask/controllers/doc_controller.py
import models.doc_model as doc_model

class DocumentNotFoundError(Exception):
    pass

def get_all_docs():
    return doc_model.get_all_docs()

def get_doc_by_id(doc_id):
    doc = doc_model.get_doc_by_id(doc_id)
    if not doc:
        raise DocumentNotFoundError('Document not found')
    return doc

def get_sections_for_document(doc_id):
    return doc_model.get_sections_for_document(doc_id)

def get_all_sections():
    return doc_model.get_all_sections()

def get_links_for_document(doc_id):
    return doc_model.get_links_for_document(doc_id)

def get_all_content(doc_id):
    """
    Get all sections with their associated links for a document.
    Returns sections with embedded links array for each section.
    """
    # Get all sections for the document
    sections = get_sections_for_document(doc_id)
    
    # Get all links for the document  
    links = get_links_for_document(doc_id)
    
    # Create a map of links by source section ID for efficient lookup
    links_by_source = {}
    for link in links:
        source_id = link.get('sourceSectionId')
        if source_id:
            if source_id not in links_by_source:
                links_by_source[source_id] = []
            links_by_source[source_id].append({
                'linkId': link.get('id'),
                'sourceSectionId': link.get('sourceSectionId'),
                'targetSectionId': link.get('targetSectionId'),
                'targetSectionTitle': link.get('targetSectionTitle'),
                'targetDocumentId': link.get('targetDocumentId'),
                'targetDocumentTitle': link.get('targetDocumentTitle'),
                'type': link.get('linkType')
            })
    
    # Build the aggregated response
    result = []
    for section in sections:
        section_id = section.get('id')
        section_data = {
            'sectionTitle': section.get('title'),
            'sectionId': section_id,
            'sectionNumber': section.get('sectionNumber'),
            'content': section.get('content'),
            'links': links_by_source.get(section_id, [])
        }
        result.append(section_data)
    
    return result